mne/channels/montage.py

```diff
@@ -1682,6 +1744,101 @@ def read_polhemus_fastscan(fname, unit='mm'):
     return points
 
 
+def _read_eeglab_locations(fname, unit):
+    # VALID_SCALES = dict(mm=1e-3, cm=1e-2, m=1)
+    # _scale = _check_unit_and_get_scaling(unit, VALID_SCALES)
+
+    ch_names = np.genfromtxt(fname, dtype=str, usecols=3).tolist()
+    topo = np.loadtxt(fname, dtype=float, usecols=[1, 2])
+    sph = _topo_to_sph(topo)
+    pos = _sph_to_cart(sph)
+    pos[:, [0, 1]] = pos[:, [1, 0]] * [-1, 1]
+
+    return ch_names, pos
+
+
+def read_standard_montage(fname, head_size=HEAD_SIZE_DEFAULT, unit='m'):
+    """Read a standard montage file containing polar coordinates.
```

Comment:
can you check if all files you can read now are in polar coordinates?

Reply:
The right way to split them would had been well possed formats vs ill-posed formats. In other wourds, if we have xyz or r,theta,phi is well posed + ch_names and types. (and we should allow for head_size=None and recover whatever is in the file with 0 magic). And this should be some kind of reader (brainvision, hpts, ..)

and `read_standard_montage` should be used for illposed ie not having radius.
But since we were ignoring it everywhere in read_montage. I just decided to stuff everything in `read_standard_montage` and drive the action with head_size.